api/src/services/assessmentsDummyService.ts

```diff
@@ -0,0 +1,185 @@
+import db from './db';
+
+interface ProgramParticipantsRow {
+  id: number;
+  principal_id?: number;
+  program_id?: number;
+  role_id?: number;
+}
+
+/**
+ * Return  program_participants table for any rows matching this principalId and this programId.
+ *
+ * @param {number} principalId - The given principalId
+ * @param {number} programId - The given programId
+ *
+ * @returns {ProgramParticipantsRow}
+ */
+export const insertToProgramParticipants = async (
+  principalId: number,
+  programId: number,
+  roleId: number
+): Promise<ProgramParticipantsRow> => {
+  // select from the program_participants table for any rows matching this principalId and this programId
+
+  let matchingProgramParticipantsRows: ProgramParticipantsRow[] =
+    await db<ProgramParticipantsRow>('program_participants')
+      .select('id', 'principal_id', 'program_id', 'role_id')
+      .where({ principal_id: principalId, program_id: programId });
+
+  // - if a row exists, update the roleId to the roleId that's being passed in, using the id of the row returned from that table as the key for updating
+
+  if (matchingProgramParticipantsRows.length) {
+    await db('program_participants')
+      .update({
+        program_id: programId,
+        role_id: roleId,
+      })
+      .where({ principal_id: principalId });
+  }
+  // - if no row exists, insert a row into the table for this principalId, programId, and roleId
+  else {
+    await db('program_participants').insert({
+      principal_id: principalId,
+      program_id: programId,
+      role_id: roleId,
+    });
+  }
+
+  matchingProgramParticipantsRows = await db<ProgramParticipantsRow>(
+    'program_participants'
+  )
+    .select('id')
+    .where({ principal_id: principalId, program_id: programId });
+
+  const insertedRowParsed = {
+    id: matchingProgramParticipantsRows[0].id,
+    principal_id: principalId,
+    program_id: programId,
+    role_id: roleId,
+  };
+
+  return insertedRowParsed;
+};
+
+interface AssessmentSubmissionRow {
+  id: number;
+  assessment_id?: number;
+  principal_id?: number;
+}
+
+interface Response {
+  id?: number;
+  assessment_id: number;
+  submission_id: number;
+  question_id: number;
+  answer_id?: number;
+  response?: string;
+  score?: number;
+  grader_response?: string;
+}
+
+export const insertToAssessmentSubmissions = async (
```

Comment:
Function header comment here please.

Reply:
Do you mean change the name for something more suitable ?